app/controllers/devise_recognisable/sessions_controller.rb

```diff
@@ -10,12 +10,12 @@ def perform_recognition_check
     self.resource = resource_class.find_by(email: params[resource_name][:email])
     return unless self.resource
 
-    previous_session = DeviseRecognisable::RecognisableSession
+    previous_sessions = DeviseRecognisable::RecognisableSession
       .where( recognisable: self.resource )
-      .order(created_at: :desc).first
-    return unless previous_session.present?
+      .order(created_at: :desc)
+    return if previous_sessions.none?
 
-    unless Recogniser.recognise?(request, previous_session)
+    unless Recogniser.recognise?(request, previous_sessions)
```

Comment:
To save passing around `request` within `Recogniser`, consider doing something like:

```
Recogniser.new(request).recognise?(previous_sessions)
```

That way you can save it as an instance variable.

If you wanna get real fancy:

```
Recogniser.with(request: request).recognise?(previous_sessions)
```

And `self.with` would just call `new`. Depends on how far down them 'make Ruby read like English' rabbit hole you wanna go.